packages/components/src/search-control/index.native.js

```diff
@@ -1,96 +1,236 @@
 /**
  * External dependencies
  */
-import { TextInput, View, TouchableHighlight } from 'react-native';
+import {
+	TextInput,
+	Text,
+	View,
+	TouchableOpacity,
+	Platform,
+	useColorScheme,
+} from 'react-native';
 
 /**
  * WordPress dependencies
  */
-import { useState, useRef } from '@wordpress/element';
-import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { useState, useRef, useMemo, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { ToolbarButton } from '@wordpress/components';
+import { Button, Gridicons } from '@wordpress/components';
 import {
 	Icon,
-	cancelCircleFilled,
-	arrowLeft,
-	search as searchIcon,
+	cancelCircleFilled as cancelCircleFilledIcon,
+	arrowLeft as arrowLeftIcon,
+	close as closeIcon,
 } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import styles from './style.scss';
+import baseStyles from './style.scss';
+import platformStyles from './platform-style.scss';
+
+// Merge platform specific styles
+for ( const selector in platformStyles ) {
+	baseStyles[ selector ] = {
+		...baseStyles[ selector ],
+		...platformStyles[ selector ],
+	};
+}
```

Comment:
I'm a bit concerned about modifying the imported `baseStyles` because it will modify the styles for the rest of the components that might import them, what do you think about creating a new constant?

Reply:
Good catch. Changing now.